src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -59,7 +59,8 @@ private async Task<Solution> MakeStaticAsync(Document document, SyntaxNode root,
 
             // Update definition to add static modifier.
             var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
-            var madeStatic = syntaxGenerator.WithModifiers(node, DeclarationModifiers.Static).WithAdditionalAnnotations(s_annotationForFixedDeclaration);
+            var oldModifiersAndStatic = syntaxGenerator.GetModifiers(node).WithIsStatic(true);
```

Comment:
For the 2nd test I have added (with `unsafe` modifier), the variable `oldMofifiersAndStatic` correctly contains both `unsafe` and `static` modifiers but when applied through `syntaxGenerator.WithModifiers(node, oldModifiersAndStatic)` the `unsafe` modifier is lost and I cannot understand why.

Reply:
This might be a bug in Rosyln's syntax generator. Would you be able to create a unit test in http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Workspaces.UnitTests/CodeGeneration/SyntaxGeneratorTests.cs,17 to verify and then file an issue on Roslyn repo: https://github.com/dotnet/roslyn/issues/new

Reply:
I have added the following test 

```csharp
VerifySyntax<MethodDeclarationSyntax>(
    Generator.MethodDeclaration("m", modifiers: DeclarationModifiers.Unsafe.WithIsStatic(true)),
    "static unsafe void m()\r\n{\r\n}");
```

which was successful so I guess there was only a problem on my usage.